content/en/blog/2019/v1beta1-authorization-policy/index.md

```diff
@@ -0,0 +1,324 @@
+| `request.headers` in constraint | `request.headers` in `when` |
+
+#### `ServiceRoleBinding`
+
+| `ServiceRoleBinding` | `AuthorizationPolicy` |
+|----------------------|-----------------------|
+| `user`  | `principals` in `from` |
+| `group` | `paths` in `to` |
+| `source.ip` in property | `ipBlocks` in `from` |
+| `source.namespace` in property | `namespaces` in `from` |
+| `source.principal` in property | `principals` in `from` |
+| `request.headers` in property | `request.headers` in `when` |
+| `request.auth.principal` in property | `requestPrincipals` in `from` or `request.auth.principal` in `when` |
+| `request.auth.audiences` in property | `request.auth.audiences` in `when` |
+| `request.auth.presenter` in property | `request.auth.presenter` in `when` |
+| `request.auth.claims` in property | `request.auth.claims` in `when` |
+
+Beyond all the differences, the `v1beta1` policy is enforced by the same engine in Envoy
+and supports the same authenticated identity (mutual TLS or JWT), condition and other
+primitives (e.g. IP, port and etc.) as the `v1alpha1` policy.
+
+See [authorization concept page](/docs/concepts/security/#authorization) for a detailed
+in-depth explanation of the `AuthorizationPolicy`.
+
+## Future of the `v1alpha1` policy
+
+The `v1alpha1` RBAC policy (`ClusterRbacConfig`, `ServiceRole`, and `ServiceRoleBinding`) is
+deprecated by the `v1beta1` authorization policy and will not be supported from Istio 1.6 onwards.
+
+Istio 1.4 and 1.5 will continue to support the `v1alpha1` RBAC policy to give you enough
+time to move away from the alpha policies.
+
+## Migration to the `v1beta1` policy
+
+Before `v1alpha1` policy is fully removed, Istio will support both `v1alpha` and `v1beta1`
+policies at the same time. However, the `v1beta1` policy overrides the `v1alpha1` policy if
+they are both applied to the same workload:
+
+1. If any `v1beta1` policies apply to the workload, the `v1beta1` policy is used and
+   the `v1alpha1` policies on the same workload will all be ignored
+1. If no `v1beta1` policy applies to the workload, the `v1alpha1` policy is used if
+   there is any
+
+{{< warning >}}
+This means when migrating to the `v1beta1`, you need to make sure the converted
+`v1beta1` policy covers all the `v1alpha1` policies applied to the same workload. The
+`v1alpha1` policies applied for the workload will be ignored after you applied the `v1beta1` policies.
+{{< /warning >}}
+
+It's recommended to migrate the `v1alpha1` policy workload-by-workload to make sure you don't
+forget any `v1alpha1` policies:
+
+1. For a given workload, find out all the `v1alpha1` policies applied to it
+1. Convert all the `v1alpha1` policies to `v1beta1` for the workload
+1. Apply the `v1beta1` policy and monitor the traffic to make sure the new policy is working as expected
+1. Continue the process for the next workload until all `v1alpha1` policies are converted
+1. Now you can remove all your `v1alpha1` policies
+
```

Comment:
What is the benefit of migrating workload by workload? How about namespace level policies? How do you migrate them per workload?

Reply:
The benefit is not to accidentally forget some policies applied to the workload.

"workload by workload" means user should think about all v1alpha1 policy applied to the workload, including mesh and namespace policies. If there is a namespace policy applying to the workload, user could convert it to a specific v1beta1 policy for the workload.

User could do this "namespace by namespace" and this is not conflict with the "workload by workload", they are on 2 different dimensions.

Reply:
I think namespace should be the main granularity. You can say the same applies to workload level. If a customer has hundreds of workloads in the mesh, and he/she certainly do not want to do "workload-by-workload" as recommended here.

Reply:
The namespace is an abstract concept which means the work needed to do is the same no matter you say "migrate namespace by namespace" or "migrate workload by workload".
There could be either 1 or 1000 workloads in a namespace, in either case the customer have to do the migration for each workload individually, there is no magic command to migrate all workloads in a namespace.

Here we emphasis on "workload" just to make sure we're on the safe side that all policies for a workload should be migrated at the same time because this is the actual boundary we support in pilot.

Updated a little bit to make this clear.

Reply:
Or maybe you can say, "It's recommended to test your new policies after you migrate your `v1alpha1` policy to make sure it works as expected. A typical flow would be: (list your steps....)"

Reply:
I rewrite this section completely, please take a look at the new contents.